corehq/ex-submodules/casexml/apps/phone/data_providers/case/livequery.py

```diff
@@ -45,8 +45,25 @@ def do_livequery(timing_context, restore_state, async_task=None):
     def index_key(index):
         return '{} {}'.format(index.case_id, index.identifier)
 
+    def is_extension(case_id):
+        """Determine if case_id is an extension case
+
+        A case that is both a child and an extension is not an extension.
+        """
+        return case_id in hosts_by_extension and case_id not in parents_by_child
+
     def has_live_extension(case_id, cache={}):
```

Comment:
yes, that's [exactly what I meant to do](http://python-guide-pt-br.readthedocs.io/en/latest/writing/gotchas/#when-the-gotcha-isn-t-a-gotcha). Python [evaluates default function parameters at the time when the function is defined](http://python-guide-pt-br.readthedocs.io/en/latest/writing/gotchas/#mutable-default-arguments).